cli/azd/pkg/account/subscriptions_manager.go

```diff
@@ -173,43 +187,66 @@ func (m *SubscriptionsManager) ListSubscriptions(ctx context.Context) ([]Subscri
 		return nil, fmt.Errorf("listing tenants: %w", err)
 	}
 
-	allSubscriptions := []Subscription{}
-	errors := []error{}
-	oneSuccess := false
-
-	for _, tenant := range tenants {
-		tenantId := *tenant.TenantID
-		subscriptions, err := m.service.ListSubscriptions(ctx, tenantId)
-		if err != nil {
-			errorMsg := err.Error()
-			displayName := *tenant.DisplayName
-			if strings.Contains(errorMsg, "AADSTS50076") {
-				errors = append(
-					errors,
-					fmt.Errorf(
+	span.SetAttributes(fields.AccountSubscriptionsListTenantsFound.Int(len(tenants)))
+
+	listForTenant := func(
+		jobs <-chan armsubscriptions.TenantIDDescription,
+		results chan<- tenantSubsResult,
+		service *azcli.SubscriptionsService) {
+		for tenant := range jobs {
+			azSubs, err := service.ListSubscriptions(ctx, *tenant.TenantID)
+			if err != nil {
+				errorMsg := err.Error()
+				displayName := *tenant.DisplayName
+				if strings.Contains(errorMsg, "AADSTS50076") {
+					err = fmt.Errorf(
 						"%s requires Multi-Factor Authentication (MFA). "+
 							"To authenticate, login with `azd login --tenant-id %s`",
 						displayName,
-						*tenant.DefaultDomain))
-			} else {
-				errors = append(
-					errors,
-					fmt.Errorf("failed to load subscriptions from tenant '%s' : %w", displayName, err))
+						*tenant.DefaultDomain)
+				} else {
+					err = fmt.Errorf("failed to load subscriptions from tenant '%s' : %w", displayName, err)
+				}
 			}
 
+			results <- tenantSubsResult{toSubscriptions(azSubs, *tenant.TenantID), err}
+		}
+	}
+
+	numJobs := len(tenants)
+	jobs := make(chan armsubscriptions.TenantIDDescription, numJobs)
+	results := make(chan tenantSubsResult, numJobs)
+	// Apply max number of concurrent workers at 50
```

Comment:
Just wondering - how did you arrive at a limit of 50?  I looked briefly at https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling to see if it might have some indication as this as a best practice but it doesn't seem to mention anything about concurrent requests.

Not pushing back on the choice, just more interested in the thought behind it so I can learn myself (I figure you might have a lot of internal Azure information here, given that you were on the other side of the control plane in a previous life ;-))

Reply:
@ellismg It was chosen as a sane default but a little arbitrarily I'm afraid. Service throttling might not be documented correctly in a lot of places, so to be safe, I came up with 50 request per second -- I wouldn't think any service would 50 rps per user as DOS. I also think 50 request per second would provide a reasonable experience for any user. Having an upper bound also avoids problems of running on older machines (I'm not sure how well `golang` handles high number of goroutines in environments with low number of logical processors).

Reply:
50 feels high and I would be concerned with exhausting outgoing threads/connections from the client vs anything on the server side.  Perhaps this number could be derived from the number of cores/threads available on the current machine.

Reply:
@wbreza I'll lower it to `25`, and make it possible for an env variable override (as an escape hatch in any case). `goroutine`s do not spin up threads, instead run on a threadpool. With TCP keep-alive, I believe the underlying `http` transport actually multiplexes the HTTP request/responses over the connection, and thus you don't end up saturating connections. Even 50 connections is considered low on any modern machine. As with any parallelism / performance questions, without real-world testing, we're making statements that are hard to prove one way or another.

Applying parallel scaling based on core(s) is not a great idea in practice. As transistor counts scaling changes overtime, your program ends up breaking subtly based on parallelism. The best idea to apply a fixed scaling, and update this when hardware has evolved enough. I don't remember the exact attribution for this advice, but I've also observed this in practice in production code.